src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -76,6 +54,65 @@ public HttpApiHandler(DurableTaskExtension config, TraceWriter traceWriter)
             return response;
         }
 
+
+        internal async Task<HttpResponseMessage> CreateCheckStatusResponse(
+            HttpRequestMessage request,
+            string instanceId,
+            OrchestrationClientAttribute attribute,
+            int totalTimeout,
+            int retryTimeout)
+        {
+            GetClientResponseLinks(request, instanceId, attribute, out var statusQueryGetUri, out var sendEventPostUri,
+                out var terminatePostUri);
+
+            if (totalTimeout < retryTimeout)
+            {
+                throw new InvalidOperationException($"Total timeout {totalTimeout} should be bigger than {retryTimeout}");
+            }
+
+            var iterationCount = totalTimeout / retryTimeout;
+            JToken durableFunctionOutput = null;
+
+            var client = GetClient(request);
+            var status = await client.GetStatusAsync(instanceId);
+            for (var i = 0; i < iterationCount; i++)
+            {
+                Thread.Sleep(retryTimeout);
+                if (status != null && status.RuntimeStatus == OrchestrationRuntimeStatus.Completed)
+                {
+                    durableFunctionOutput = status.Output;
+                    break;
+                }
+                status = await client.GetStatusAsync(instanceId);
+            }
+
+            if (durableFunctionOutput == null)
```

Comment:
If the orchestration completes but returns a `null` output, this code will return a 202 response, which is not correct. Rather than using the variable for the output, you should use a separate variable for if the operation timed-out.